nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/operation/HotReloadStudioSnapshot.java

```diff
@@ -71,46 +80,110 @@
     protected boolean validate = true;
 
     @OperationMethod
-    public void run() throws Exception {
+    public Blob run() {
+
+        JSONArray result = new JSONArray();
+        JSONObject resultJSON = new JSONObject();
+
         if (updateInProgress) {
-            return;
+            resultJSON.put("status", "inProgress");
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         if (!((NuxeoPrincipal) session.getPrincipal()).isAdministrator()) {
-            throw new NuxeoException("Must be Administrator to use this function");
+            resultJSON.put("status", "error");
+            resultJSON.put("message", "Must be Administrator to use this function");
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         if (!Framework.isDevModeSet()) {
-            throw new NuxeoException("You must enable Dev mode to Hot reload your Studio Snapshot package.");
+            resultJSON.put("status", "error");
+            resultJSON.put("message", "You must enable Dev mode to Hot reload your Studio Snapshot package.");
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         List<DownloadablePackage> pkgs = pm.listRemoteAssociatedStudioPackages();
         DownloadablePackage snapshotPkg = StudioSnapshotHelper.getSnapshot(pkgs);
 
         if (snapshotPkg == null) {
-            throw new NuxeoException("No Snapshot Package was found.");
+            resultJSON.put("status", "error");
+            resultJSON.put("message", "No Snapshot Package was found.");
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         try {
             updateInProgress = true;
-            hotReloadPackage(snapshotPkg);
+            hotReloadPackage(snapshotPkg, resultJSON);
```

Comment:
Don't give your result as arguments (especially in this case where the `resultJSON` variable is empty). Consider returning the `resultJSON` or directly the created `Blob`.